lib/batch/subject.csv.rb

```diff
@@ -16,24 +19,28 @@ def add_metadata_to_csv(model)
   query.each.with_index do |gf, idx|
     gfmod = gf.each.map {|k, v| [k.gsub(/_[a-z]+\z/, ''), v] }.to_h
 
-    csv << terms.map {|term|
+    row = @terms.map {|term|
       t = term.to_s
 
-      if t == "events"
-        events = model.constantize.find(gfmod["id"]).events
-        events.first.blank? ? "no event" : events.first[:action]
-      elsif gfmod[t].is_a?(Array)
+      if gfmod[t].is_a?(Array)
         gfmod[t].reject(&:blank?).compact.join(' ; ')
       else
         gfmod[t].to_s
       end
     }
 
+    record = model.constantize.find(gfmod["id"])
+    event = record.events.first.blank? ? "no event" : record.events.first[:action]
+    row << event
+    row << record.collection_ids.reject(&:blank?).compact.join(' ; ')
```

Comment:
@Meowcenary suggestion: 
```suggestion
    row << record.collection_ids.reject(&:blank?).join(' ; ')
```